nixos/modules/services/networking/tetrd.nix

```diff
@@ -0,0 +1,46 @@
+{ config, lib, pkgs, ... }:
+
+let
+  tetrd-init = pkgs.writeShellScript "tetrd-init" ''
+    mkdir -p /usr/bin
+    ln -sf ${pkgs.nettools}/bin/{route,ifconfig} /usr/bin
+  '';
+in {
+  options.services.tetrd.enable = lib.mkEnableOption pkgs.tetrd.meta.description;
+
+  config = lib.mkIf config.services.tetrd.enable {
+    environment.systemPackages = [ pkgs.tetrd ];
+
+    systemd.services.tetrd = {
+      description = pkgs.tetrd.meta.description;
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        ExecStartPre = tetrd-init;
+        ExecStart = "${pkgs.tetrd}/opt/Tetrd/bin/tetrd";
+        Restart = "always";
+        RuntimeDirectory = "tetrd";
+        RootDirectory = "/run/tetrd";
+        DynamicUser = true;
+        PermissionsStartOnly = true;
+        BindReadOnlyPaths = [ builtins.storeDir ];
+
+        CapabilityBoundingSet = [
+          "CAP_DAC_OVERRIDE"
+          "CAP_NET_ADMIN"
+        ];
+
+        AmbientCapabilities = [
+          "CAP_DAC_OVERRIDE"
+          "CAP_NET_ADMIN"
+        ];
+
+        BindPaths = [
+          "/etc"
+          "/run"
+          "/var"
```

Comment:
Does it need write access to `/etc` and the whole `/var`?

Reply:
For /var, it's just /var/log and /var/run, I'm not entirely sure about /etc other than resolv.conf, I'll have to investigate.